lib/transport/selenium-webdriver/index.js

```diff
@@ -269,7 +269,7 @@ class Transport extends BaseTransport {
     const startTime = new Date();
     const {host, port, start_process} = this.settings.webdriver;
     const portStr = port ? `port ${port}` : 'auto-generated port';
-    const options = this.createSessionOptions(argv);
+    const options = await this.createSessionOptions(argv);
```

Comment:
createSessionOptions doesn't return a Promise.

Reply:
It does now, for AppAutomate, since I am uploading the app to BrowserStack inside this method only, after creating all the session options. It also made more sense to do it inside `createSessionOptions` since `options` are not final until the app is uploaded to BrowserStack. And for the other implementations of `createSessionOptions`, adding `await` here shouldn't have any effect.

I tried doing this inside `this.createDriver` also (just so that I don't have to add this `await`) but the problem there was that the spinner (Connecting to localhost on 4723...) was overwriting the console messages related to app upload (Uploading app to BrowserStack... and App upload successful!) and the spinner is called just after calling `this.createSessionOptions` method.